.eslintrc

```diff
@@ -24,12 +25,28 @@
     "rules": {
         "valid-jsdoc": 2,
         "no-restricted-globals": [
-            "error",
+            "warn",
             {
                 "name": "VTTCue",
-                "message": "import VTTCue from 'parsers/captions/vttcue'"
+                "message": "import VTTCue from 'os/parsers/captions/vttcue'"
+            },
+            {
+                "name": "window",
+                "message": "Use `self` instead of `window` to access the global context everywhere (including workers)."
+            },
+            {
+                "name": "document",
+                "message": "We're using the DOM here."
+            },
+            {
+                "name": "navigator",
```

Comment:
This much. But this rule should actually be changed to an error and these issues addressed - the exception being that I will remove the `window` rule because we don't need to use `self` and there's some risk that sloppy websites override the `self` global.
```
/dev/jwplayer/src/js/api/config.js
  78:41  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/api/core-bundle-loader.js
  41:39  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  43:37  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/api/core-shim.js
  330:10  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  333:30  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/api/get-media-element.js
  8:17  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/api/setup-steps.js
  82:16  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/compatibility.js
  177:40  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  351:9   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  353:3   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/controller/controller.js
  270:43  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  270:56  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  272:29  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  272:67  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  274:41  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/controller/instream-adapter.js
  171:29  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/controller/tracks-loader.js
  68:50  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/environment/environment.ts
  30:27  warning  Unexpected use of 'navigator'. We're using the browser here  no-restricted-globals

/dev/jwplayer/src/js/jwplayer.js
  42:30  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/model/storage.js
  9:15  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/parsers/captions/vttcue.js
   17:14  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  265:24  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  266:43  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/plugins/plugin.js
  35:41  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/plugins/plugins.ts
  20:5   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  30:16  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/plugins/utils.ts
  17:17  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/polyfills/promise.js
  3:16  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  3:35  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/polyfills/webvtt.js
  961:6  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  962:5  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/program/media-element-pool.ts
  100:26  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/providers/bandwidth-monitor.ts
  17:40  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/providers/html5.js
   42:22  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  547:31  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  602:31  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals

/dev/jwplayer/src/js/providers/tracks-mixin.ts
  515:25  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  515:42  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  828:23  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/providers/utils/network-error-parser.js
  13:58  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/utils/active-tab.ts
  2:21  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  4:21  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  7:27  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  9:21  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/utils/ajax.js
  47:29  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  49:48  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/backbone.events.ts
  72:15  warning  Unexpected aliasing of 'this' to local variable  @typescript-eslint/no-this-alias

/dev/jwplayer/src/js/utils/browser.ts
   3:19  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  18:26  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  18:63  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  76:16  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  76:32  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  87:21  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  97:16  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  99:25  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/can-autoplay.js
  29:53  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  32:14  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/clock.ts
  4:50  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/css.js
   10:20  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  115:24  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/utils/dom.js
    5:19  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   28:22  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   72:17  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  194:18  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  197:5   warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  217:22  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  222:27  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  223:27  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  253:13  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  260:90  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  267:17  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  272:28  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  272:57  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/helpers.ts
  47:15  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  48:15  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/utils/language.js
  78:16  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals
  80:36  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  84:40  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  87:24  warning  Unexpected use of 'navigator'. We're using the browser here                                                             no-restricted-globals

/dev/jwplayer/src/js/utils/parser.ts
   7:16  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  50:26  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/random-id-generator.ts
  6:24  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  6:41  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/request-animation-frame.ts
  1:82  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  3:63  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/scriptloader.js
  14:18  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  22:23  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  91:26  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  91:70  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/utils/strings.ts
  3:20  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/ui.js
    7:42  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
    8:47  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  217:42  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  248:26  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  313:41  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  365:62  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/underscore.js
  46:24  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/validator.ts
  5:18  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/utils/video.ts
  1:37  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/captionsrenderer.js
   43:16  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   77:33  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  127:27  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  128:26  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  239:36  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/button.ts
  22:21  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/chapters.mixin.js
  8:19  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/custom-button.js
  22:31  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  35:27  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/menu/category-button.ts
  41:29  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/menu/menu.js
  60:37  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/menu/settings-menu.js
  310:9  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  318:9  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  373:9  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/simple-tooltip.js
  4:28  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  8:25  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/slider.js
  12:25  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  13:38  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/timeslider.js
   15:21  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
   17:20  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
   23:25  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  310:13  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/components/tooltipicon.js
   9:19  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  18:24  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/controlbar.js
   16:21  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   25:21  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   34:21  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   50:34  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   51:30  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  108:41  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  242:19  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  378:17  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  378:55  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals

/dev/jwplayer/src/js/view/controls/controls.js
  333:25  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  420:57  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/info-overlay.js
   82:13  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   94:13  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  122:16  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/view/controls/rightclick.js
  185:9   warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  189:13  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  214:9   warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  215:9   warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/controls/shortcuts-tooltip.js
  79:9  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals
  87:9  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/floating-drag-ui.js
  28:31  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  29:30  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/view/preview.ts
  46:28  warning  Unexpected use of 'document'. We're using the DOM here  no-restricted-globals

/dev/jwplayer/src/js/view/utils/views-manager.js
    9:36  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
    9:63  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   16:34  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   80:5   warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   81:5   warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   84:9   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   87:5   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   88:9   warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   89:9   warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
   90:9   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
   93:13  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  108:13  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/view/utils/visibility.js
  24:14  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  24:29  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  32:18  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  33:18  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  47:9   warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  65:20  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals

/dev/jwplayer/src/js/view/view.js
  109:23  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  247:69  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  311:23  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  373:33  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  374:33  warning  Unexpected use of 'window'. Use `self` instead of `window` to access the global context everywhere (including workers)  no-restricted-globals
  435:45  warning  Unexpected use of 'document'. We're using the DOM here                                                                  no-restricted-globals
  729:15  warning  Unexpected use of 'document'. We're using the DOM here
```